eth_tester/utils/backend_testing.py

```diff
@@ -474,12 +486,112 @@ def test_estimate_gas(self, eth_tester):
             math_address,
             'increment',
         )
-        gas_estimation = eth_tester.estimate_gas(dissoc(estimate_call_math_transaction, 'gas'))
+        if not is_pyevm_available():  # py-evm requires the gas parameter
+            estimate_call_math_transaction = dissoc(estimate_call_math_transaction, 'gas')
+        gas_estimation = eth_tester.estimate_gas(estimate_call_math_transaction)
         call_math_transaction = assoc(estimate_call_math_transaction, 'gas', gas_estimation)
         transaction_hash = eth_tester.send_transaction(call_math_transaction)
         receipt = eth_tester.get_transaction_receipt(transaction_hash)
         assert receipt['gas_used'] == gas_estimation
 
+    def test_can_call_after_exception_raised_calling_pyethereum(self, eth_tester):
+        self.skip_if_no_evm_execution()
+        if is_pyevm_available():
+            pytest.skip('Test only relevant for pyethereum.')
+        elif is_pyethereum16_available():
+            from ethereum.tester import TransactionFailed
+        elif is_pyethereum21_available():
+            from ethereum.tools.tester import TransactionFailed
+
+        throws_address = _deploy_throws(eth_tester)
+        call_will_throw_transaction = _make_call_throws_transaction(
+            eth_tester,
+            throws_address,
+            'willThrow',
+        )
+        with pytest.raises(TransactionFailed):
+            eth_tester.call(call_will_throw_transaction)
+
+        call_value_transaction = _make_call_throws_transaction(
+            eth_tester,
+            throws_address,
+            'value',
+        )
+        raw_result = eth_tester.call(call_value_transaction)
+        result = _decode_throws_result('value', raw_result)
+        assert result == (1,)
+
+    def test_can_call_after_exception_raised_calling_pyevm(self, eth_tester):
```

Comment:
It is a bug that the py-evm backend doesn't raise the general `TransactionFailed` exception, too. But it may be too long of a fix to include in this PR (I don't know until I take a look later).

If so: you can leave this test as `pyevm`-specific until that bug is fixed. Just comment in this pyevm-specific test that "this is testing **broken** behavior, and when py-evm is fixed, then this test should be deleted, and we should remove the pyevm skip in the general version of the test above."

Reply:
Sounds good.